### PR TITLE
New Data Source : `azurerm_vpn_server_configuration`

### LINKED ISSUE #26989: Support for a azurerm_vpn_server_configuration datasource
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

The azurerm_vpn_server_configuration resource is typically managed and associated with the Azure Virtual WAN.
When I'm creating VPNs in a Virtual Hub i need to know the ID of the proper azurerm_vpn_server_configuration that I want to use.  Usually Creating WANs & VPN Server Configurations is done separately from creating Virtual Hubs ( at least in my world )

### New or Affected Resource(s)/Data Source(s)

azurerm_vpn_server_configuration

### Potential Terraform Configuration

```hcl
data "azurerm_vpn_server_configuration" "example" {

  name                     = "example-VPNSC"
  resource_group_name      = "example-rg"


}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by sinbai**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

Support for a `azurerm_vpn_server_configuration` data source.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

```
PASS: TestAccVPNServerConfigurationDataSource_azureAD (289.43s)
PASS: TestAccVPNServerConfigurationDataSource_certificate (228.86s)
PASS: TestAccVPNServerConfigurationDataSource_radius (225.54s)
PASS: TestAccVPNServerConfigurationDataSource_tags (210.99s)
```

image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* New Data Source : `azurerm_vpn_server_configuration` [GH-26989]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #26989

### CODE REVIEW

File: internal/services/network/vpn_server_configuration_data_source_resource.go [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/network/2023-11-01/virtualwans"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+func dataSourceVPNServerConfiguration() *pluginsdk.Resource {
```
Comment (stephybun) [inline internal/services/network/vpn_server_configuration_data_source_resource.go:21]: @sinbai we're not accepting new untyped resource or data sources. Can you please convert this to a typed data source before we proceed with the review here?

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @sinbai, please see the comment left in-line

File: internal/services/network/vpn_server_configuration_data_source_resource.go (hunk 1 above)
Comment (sinbai) [reply internal/services/network/vpn_server_configuration_data_source_resource.go]: Fixed.

**Review by katbyte — APPROVED** [review]
LGTM 🌷

**Pull request merged**